File: Beakjoon_Solved.ac/Class3/BJ_Q1107-리모컨/solution.java

import java.io.*;
import java.util.StringTokenizer;

public class solution {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());
        int ans1;
        int ans2;
        if(m == 0){
            ans1 = (Integer.toString(n)).length();
            ans2 = (n>=100?n-100:100-n);
            System.out.println(ans1>ans2?ans2:ans1);
        }
        else{
            StringTokenizer st = new StringTokenizer(br.readLine());
            Integer[] nlist = new Integer[m];
            int tmp;
            int jud;
            for(int i = 0 ;i<m; i++){
                nlist[i] = Integer.parseInt(st.nextToken());
            }
            if(n == 100){
                System.out.println(0);
            }
            else if(n > 100){
                ans1 = n -100;
                ans2 = 2*n;
                tmp = ans2;
                for(int i = 0; i < ans2; i++){
                    String strI = Integer.toString(i);
                    jud = 0;
                    for(int j : nlist){
                        for(int k = 0; k < strI.length(); k++){
                            if(Character.getNumericValue(strI.charAt(k)) == j){
                                jud = 1;
                                break;
                            }
                        }
                        if(jud == 1){
                            break;
                        }
                    }
                    if(jud == 0){
                        if(i<=n){
                            tmp = n - i + strI.length();
                        }
                        else{
                            tmp = i - n + strI.length();
                        }
                    }
                    if(tmp < ans2){
                        ans2 = tmp;
                    }
                }
                System.out.println("ans1 : " + ans1+ " ans2 : " + ans2);
                System.out.println((ans1<ans2 ? ans1 : ans2));
            }
            else{ // n < 100
                ans1 = 100 - n;
                ans2 = 100 + n;
                tmp =  ans2;

                for(int i = 0; i < ans2; i++){
                    String strI = Integer.toString(i);
                    jud = 0;
                    for(int j :nlist){
                        for(int k = 0; k < strI.length(); k++){
                            if(Character.getNumericValue(strI.charAt(k)) == j){
                                jud = 1;
                                break;
                            }
                        }
                        if(jud == 1){
                            break;
                        }
                    }
                    if(jud == 0){
                        if(i<=n){
                            tmp = n - i + strI.length();
                        }
                        else{
                            tmp = i - n + strI.length();
                        }
                    }
                    if(tmp < ans2){
                        ans2 = tmp;
                    }

                }
                System.out.println("ans1 : " + ans1+ " ans2 : " + ans2);
                System.out.println((ans1<ans2 ? ans1 : ans2));

            }

            
        }

    }
}

File: Beakjoon_Solved.ac/Class3/BJ_Q1107-리모컨/solution.py
n = int(input())
m = int(input())
nlist = []
ans = 0
if(m == 0):
    ans1 = len(str(n))
    ans2 = ( n-100 if(n>=100) else 100-n)
    print(ans1 if(ans1<ans2) else ans2)
else:
    nlist = input().split(" ")
    if(n==100):
        print(0)
    elif(n>100):
        ans1 = n - 100
        ans2 = n + 100
        tmp = ans2
        for i in range(100, 1000000):
            jud = 0
            for j in nlist:
                if(j in str(i)):
                    jud = 1
                    break
            if(jud == 0):
                if(i<=n):
                    tmp = n - i + len(str(i))
                else:
                    tmp = i - n + len(str(i))
            if(tmp < ans2):
                ans2 = tmp
        ans = (ans1 if(ans1<ans2) else ans2)
        print(ans)
        
    else:
        ans1 = 100 - n # 100 에서 이동
        ans2 = 100 + n
        tmp = ans2
        cnt =0
        for i in range(0, 201):
            jud = 0
            for j in nlist:
                if(j in str(i)):
                    jud = 1 # 고장난 번호가 포함되어 있다면
                    break
                    
            if(jud == 0): # 고장난 번호가 포함되어 있지 않다면
                if(i>n):
                    tmp = len(str(i)) + i - n
                elif(i<n):
                    tmp = n - i +len(str(i))
                elif(i==n):
                    tmp = len(str(i))
            if(tmp < ans2):
                ans2 = tmp
                
                
        ans = (ans1 if(ans1<ans2) else ans2)
        print(ans)
            
    